server/middleware/torCheckMiddleware.js

```diff
@@ -0,0 +1,24 @@
+const logger = require('winston');
+const db = require('../models');
+
+const torCheck = (req, res, next) => {
+  const { ip } = req;
+  logger.debug(`tor check for: ${ip}`);
+  return db.Tor.findAll(
+    {
+      where: {
+        address: ip,
+      },
+      raw: true,
+    })
+    .then(result => {
+      logger.debug('tor check results:', result);
+      req['tor'] = (result.length >= 1); // add this to the req object
```

Comment:
Maybe you should just stop the request here? Not sure, if there are other routes in that may be blocked in the future it will be simpler to add. If it's gonna be only the /publish route then this is probably fine.